### PR TITLE
[Chore] Add unit test for http api client - login

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 👀

Start to add unit test at the ApiClient - network level.
 
## Insight 📝

- Add @visibleForTesting method to inject a test behavior.
- Insert mock response, alter response body.
- Test the behavior when using HttpClient() from HttpClientProvider (HTTP API) when calling to `login()` on success/failed case.
 
## Proof Of Work 📹
💁‍♂️  codecov++

### CODE REVIEW

File: test/api/http/api_client_test.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        'client_id': 'client_id',
+        'client_secret': 'client_secret'
+      }));
+
+      expect(result, isA<AuthTokenResponseData>());
+      expect(result.data.type, 'token');
+      expect(result.data.authToken.accessToken, 'access token');
+    });
+
+    test('When login unsuccessfully, it returns DioError', () async {
```
Comment (manh-t) [inline test/api/http/api_client_test.dart:48]: `When login failed` 🤪

File: lib/api/http/api_client_provider.dart [hunk 2]
```diff
@@ -23,6 +23,12 @@ class ApiClientProvider {
     return ApiClientProvider._(dio);
   }
 
+  // A utility method for unit testing, developers can alter the response as expected
+  @visibleForTesting
+  void injectHttpClientAdapter(HttpClientAdapter httpClientAdapter) {
```
Comment (manh-t) [inline lib/api/http/api_client_provider.dart:28]: 👍

**Review by manh-t — APPROVED** [review]

File: test/mocks/generate_mocks.dart [hunk 3]
```diff
@@ -10,7 +11,8 @@ import 'package:survey/use_cases/get_profile_use_case.dart';
   OAuthRepository,
   GraphQLClient,
   GetProfileUseCase,
-  ApiClient
+  ApiClient,
+  HttpClientAdapter
```
Comment (luongvo) [inline test/mocks/generate_mocks.dart:15]: 👍

**Review by luongvo — APPROVED** [review]
sweet

**Pull request merged**